### ISSUE TITLE
Slow progress using accelerate on multi-node

### ISSUE THREAD

**Issue by haukened** [lucidrains/imagen-pytorch#214]
When using multi-GPU via the `accelerate` scripts, performance is improved, however when doing multi-node-multi-GPU performance degrades below usability.

Benchmarks:

1. Single P4 GPU: 1.8 it/sec
2. Dual P4 GPU (Same host): 2.6 it/sec
3. Quad P4 GPU (Two hosts): 285.15 sec/it

`Iteration:   2%|██▊                   | 12/500 [32:55<38:39:10, 285.15s/it]`

Opening this issue as a way to track resolution for public information.

**Comment by lucidrains** [comment]
@haukened ohh, thanks for reporting this

I don't have any means of testing this

@muellerzr do you have any ideas of how to diagnose what is wrong?

**Comment by haukened** [comment]
Just to cover some bases because "its always the network" 
Here's the setup:

- 2x Ubuntu 22.04 Virtual Machines, each with 4x vCPU, 32GB RAM, 2x NVIDIA Tesla P4's
- Running on two hosts in the same VMware cluster
- Connected to a Cisco Nexus 9396PX via 2x10Gbps LACP bridge each host

**Comment by muellerzr** [comment]
I'm on vacation for a few weeks, CC @sgugger to pass along :)

**Comment by lucidrains** [comment]
@muellerzr have a great vacay! I need a vacation 🤔

**Comment by haukened** [comment]
Wait. What’s a vacation?

**Comment by lucidrains** [comment]
@haukened you know, that thing we took for granted before the pandemic hit :laughing:

**Comment by deepglugs** [comment]
I was able to get reasonable results after switching my dataloader to use webdataset.  There was likely a huge bottleneck in my custom dataloader.  I was able to get 4 GPUs going, but noticed loss was 0.0 (and stays at 0.0).  I'll mess around with the learning rate and see if that helps.

**Comment by muellerzr** [comment]
@haukened two very common debug potentials.

Have we adjusted:

- The batch size
- The learning rate by n_gpus

What I mean by this is accelerate does things on a per-device standard. So to run two nodes that have a bs of 64, when working on a single GPU you should run w/ a batch size of 128 to compare performance well

And what does your `accelerate config` look like? (And version) :) Back on vacation so I can take a peek into this, just need a little more info

**Comment by deepglugs** [comment]
I did test learning rate 1e-4 and 1e-5 for 4 GPUs and still got zero loss. Should I go even lower?

Looking through the docs and trainer code, I noticed we aren't calling accelerator.accumulate().  Should we be warping our forward() in that?

**Comment by deepglugs** [comment]
I attempted to wrap trainer.forward() and trainer.update_unet() with accelerator.accumulate(), but still getting 0 loss.

Accelerate 0.12.0

My accelerate config: 
```
compute_environment: LOCAL_MACHINE
deepspeed_config: {}
distributed_type: MULTI_GPU
downcast_bf16: 'no'
fsdp_config: {}
machine_rank: 0
main_process_ip: null
main_process_port: null
main_training_function: main
mixed_precision: 'no'
num_machines: 1
num_processes: 4
use_cpu: false
```

**Comment by muellerzr** [comment]
Thanks @deepglugs. Last thing is could you show the code you're using? :)

**Comment by deepglugs** [comment]
My training loop is here: https://github.com/deepglugs/deep_imagen/blob/main/imagen.py#L662

**Comment by muellerzr** [comment]
@deepglugs might be good to open a different issue? Since this is not multi-node anymore (just multi-gpu)

**Comment by muellerzr** [comment]
@haukened if you could also provide what I asked earlier out of deepglugs (sorry didn't realize we were taking over the issue for something else!) this would help greatly :)
